### PR TITLE
feat(plugin): implement label point field

### PR DESCRIPTION
**Pull request by nourbalaha**
Done:
- implement label point field
- add functionality to color field
- create text field
- create switch field
- create select field
- refactor

### CODE REVIEW

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/commonStyles/index.ts [hunk 1]
```diff
@@ -42,17 +43,23 @@ export const FieldWrapper = styled.div<{ gap?: number }>`
...
+  justify-content: start;
+  align-items: center;
   border: 1px solid #d9d9d9;
   border-radius: 2px;
   flex: 1;
   height: 100%;
   width: 100%;
 `;
 
+export const BorderLessFieldValue = styled(FieldValue)`
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/commonStyles/index.ts:55]: BorderlessFieldValue
L doesn't need to be capitalized

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/SwitchField.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Switch } from "@web/sharedComponents";
+import { ComponentProps } from "react";
+
+import { FieldTitle, BorderLessFieldValue, FieldWrapper } from "../commonStyles";
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/SwitchField.tsx:4]: Can you rename the commonStyles folder to commonComponents? They use the styled components library, but they are components. `commonStyles` sounds like css only, which is a little misleading

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/ColorField.tsx [hunk 3]
```diff
@@ -1,20 +1,43 @@
...
+  const [selectedColor, setSelectedColor] = useState(color);
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setText(e.target.value);
+    setSelectedColor(() => {
+      const result = isValidColor(e.target.value) ? e.target.value : "";
+      onChange?.(result);
+      return result;
+    });
+  };
```
Comment (KaWaite) [inline plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/ColorField.tsx:29]: Have you tested this by manually typing in a value?
If I understand the code correctly, if I want to type "red", and I type the first r, wouldn't this revert back to ""? So I'd never be able to manually type in a color, only copy/paste

Comment (nourbalaha) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/ColorField.tsx:29]: I have tested it and it works, can you please test it as well.
Please note that the color field has two states one for the text and the other one for the selected color.

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/SwitchField.tsx (hunk 2 above)
Comment (nourbalaha) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/common/SwitchField.tsx]: Done!

File: plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/commonStyles/index.ts (hunk 1 above)
Comment (nourbalaha) [reply plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/point/commonStyles/index.ts]: Done! added `noBorder` prop to `FieldValue`.

**Review by KaWaite — APPROVED** [review]

**Pull request merged**